vulnerabilities/importers/apache_httpd.py

```diff
@@ -49,47 +54,81 @@ def updated_advisories(self):
-            advisories = to_advisories(data)
+            links = fetch_links(self.url)
+            advisories = []
+            for link in links:
+                data = requests.get(link).json()
+                advisories.append(self.to_advisory(data))
             return self.batch_advisories(advisories)
 
         return []
 
-
-def to_advisories(data):
-    advisories = []
-    for issue in data:
-        resolved_packages = []
-        impacted_packages = []
-        for info in issue:
-            if info.tag == "cve":
-                cve = info.attrib["name"]
-
-            if info.tag == "title":
-                summary = info.text
-
-            if info.tag == "fixed":
-                resolved_packages.append(
-                    PackageURL(type="apache", name="httpd", version=info.attrib["version"])
-                )
-
-            if info.tag == "affects" or info.tag == "maybeaffects":
-                impacted_packages.append(
-                    PackageURL(type="apache", name="httpd", version=info.attrib["version"])
-                )
-
-        advisories.append(
-            Advisory(
-                vulnerability_id=cve,
-                summary=summary,
-                impacted_package_urls=impacted_packages,
-                resolved_package_urls=resolved_packages,
+    def to_advisory(self, data):
+        cve = data["CVE_data_meta"]["ID"]
+        descriptions = data.get("description", {}).get("description_data", [])
+        description = None
+        for desc in descriptions:
+            if desc.get("lang") == "eng":
+                description = desc.get("value")
+                break
+
+        impacts = data.get("impact", [])
+        impact = None
+        for imp in impacts:
+            value = imp.get("other")
+            if value is not None:
+                impact = value
+                break
+
+        if impact is not None:
+            severity = VulnerabilitySeverity(
+                system=scoring_systems["apache_httpd"],
+                value=impact,
+            )
+            reference = Reference(
+                reference_id=cve,
+                url=self.ref_url.format(cve),
+                severities=[severity],
+            )
+        else:
+            reference = Reference(
+                reference_id=cve,
+                url=self.ref_url.format(cve),
             )
-        )
 
-    return advisories
+        fixed_packages = []
+        affected_packages = []
+
+        for vendor in data["affects"]["vendor"]["vendor_data"]:
+            for products in vendor["product"]["product_data"]:
+                for version in products["version"]["version_data"]:
+                    version_value = version["version_value"]
+
+                    if version["version_affected"] == "<":
+                        fixed_packages.append(
+                            PackageURL(type="apache", name="httpd", version=version_value)
+                        )
+
+                    else:
```

Comment:
This is probably right, but could you be more explicit with expected values of version range operator. 
Something like:
```python
elif version["version_affected"] == "=" or version["version_affected"] == "<=" .... : 
..... 
```

would be more helpful.

Reply:
We would also want to resolve that version  range into discrete versions, that way this data would be more rich. 

For fetching all version of httpd you'd need https://github.com/nexB/vulnerablecode/blob/deb44c68345315085fbab07d2f8b58e93327d08b/vulnerabilities/package_managers.py#L345 and https://github.com/apache/httpd . Anyways let me know if you need additional hep here.